src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -17,25 +18,15 @@
    (contains? edge "target")
    (contains? edge "relationship")))
 
-(defn resource-event?
+(pls/defn-validated resource-event? :- report/resource-event-schema
   "Returns true if it looks like a resource event"
   [event]
-  (and
-   (map? event)
-   (contains? event "status")
-   (contains? event "timestamp")
-   (contains? event "resource_title")
-   (contains? event "property")
-   (contains? event "message")
-   (contains? event "new_value")
-   (contains? event "old_value")
-   (contains? event "resource_type")))
-
-(defn report?
+  (keywordize-keys event))
```

Comment:
@senior it does that same kind of validation as before though, to a consumer of the function via pre/post conditions, it is the same. If we had consumers outside of pre and post conditions I suppose it wouldn't make much sense.